src/GraphQL.Federation.Tests/FederationFieldTests.cs

```diff
@@ -0,0 +1,77 @@
+using System.Text.Json;
+using GraphQL.DataLoader;
+using GraphQL.Federation.Tests.Schema;
+using GraphQL.SystemTextJson;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace GraphQL.Federation.Tests;
+
+// dotnet test GraphQL.Federation.Tests --filter DisplayName~GraphQL.Federation.Tests.ServiceTest
+// dotnet test GraphQL.Federation.Tests --filter DisplayName~GraphQL.Federation.Tests.EntitiesTest
+[Collection(nameof(ServiceProviderCollectionDefinition))]
+public class FederationFieldTests : BaseGraphQLTest
+{
+    public FederationFieldTests(ServiceProviderFixture serviceProviderFixture)
+        : base(serviceProviderFixture)
+    { }
+
+    [Fact]
+    public async Task ServiceTest()
+    {
+        // WaitForDebugger();
+
+        var schema = Services.GetService<TestSchema>();
+
+        var json = await schema.ExecuteAsync(new GraphQLSerializer(), options =>
+        {
+            options.Listeners.Add(new DataLoaderDocumentListener(new DataLoaderContextAccessor()));
+            options.ThrowOnUnhandledException = true;
+            options.Query = @"
+query {
+    _service {
+        sdl
+    }
+}";
+        }).ConfigureAwait(false);
+
+        var sdl = JsonDocument.Parse(json).RootElement
+            .GetProperty("data")
+            .GetProperty("_service")
+            .GetProperty("sdl")
+            .GetString();
+
+        sdl.ShouldBe("schema {\n  query: TestQuery\n}\n\ntype DirectivesTestDto @key(fields: \"id\") @shareable @inaccessible {\n  id: Int!\n  shareable: String @shareable\n  inaccessible: String @inaccessible\n  override: String @override(from: \"OtherSubgraph\")\n  external: String @external\n  provides: String @provides(fields: \"foo bar\")\n  requires: String @requires(fields: \"foo bar\")\n}\n\ntype ExternalResolvableTestDto @key(fields: \"id\") {\n  id: Int!\n  external: String @external\n  extended: String! @requires(fields: \"external\")\n}\n\ntype ExternalTestDto @key(fields: \"id\", resolvable: false) {\n  id: Int!\n}\n\ntype FederatedTestDto @key(fields: \"id\") {\n  id: Int!\n  name: String\n  externalTestId: Int!\n  externalResolvableTestId: Int!\n  externalTest: ExternalTestDto!\n  externalResolvableTest: ExternalResolvableTestDto! @provides(fields: \"external\")\n}\n\ntype TestQuery {\n  directivesTest: DirectivesTestDto!\n}\n\n\nextend schema @link(url: \"https://specs.apollo.dev/federation/v2.0\", import: [\"@key\", \"@shareable\", \"@inaccessible\", \"@override\", \"@external\", \"@provides\", \"@requires\"])");
```

Comment:
There's some way to tell Shouldly to ignore line ending differences... can't remember how at the moment.  Probably another argument.

Reply:
I'll try to use `ShouldBeCrossPlat` from https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL.Tests/AssertionExtensions.cs